File: LeetCode/187/main.cpp
class Solution {
public:
  vector<string> findRepeatedDnaSequences(string s) {
    int n = s.length();
    vector<string> ans;
    unordered_map<string, int> mp;

    for (int i = 0; i <= n; i++) {
      string temp = s.substr(i, 10);
      if (mp[temp] != -1) {
        mp[temp]++;
      }
      if (mp[temp] > 1) {
        ans.push_back(temp);
        mp[temp] = -1;
      }
    }
    return ans;
  }
};

File: LeetCode/187/main.py
from collections import defaultdict


class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        n = len(s)
        mp = defaultdict(int)
        ans = []

        for i in range(n+1):
            substr = s[i:i+10]
            if (mp[substr] != -1):
                mp[substr] += 1
            if (mp[substr] > 1):
                ans.append(substr)
                mp[substr] = -1

        return ans
